### DISCUSSION TITLE
Timer in UI?

### DISCUSSION THREAD

**Discussion by yanshay** [slint-ui/slint#4385] [Q&A] [answered]
I saw Slint has Timer in the Rust API, but couldn't find something similar in the UI.

I want to handle delayed behavior in the UI, for example, when a user presses a button, after two seconds some other UI component will change appearence. I couldn't find how to do that w/o involving rust code.
I did try `animation_tick()` for that, but once I used it the application continue running the event loop even when the app is idle anf the delayed activation is not relevant, probably because I have an expression that depends on animation_tick(). This is not good for me on the backend side.
Also, I thought about using `animate` for delaying visual appearance, but there's not support for Step (that I saw is available in css), the closest is the bezier which comes close, but not perfect, and for my specific scenario I can't get it to work the way I want.

- Is there a way to achieve what I'm looking for?
- Maybe I'm missing something here, but what is the Slint timer good for when any backend has timers anyway? Why use Slint timer rather than the standard timers of Rust for example? Maybe I could use it by triggering it from a callback and having the callback call upon a function back in the UI? But it seems abit more complex than it should.

**Comment by ogoffart** [comment, marked as answer]
> I have an expression that depends on animation_tick().

True, this will cause the expression to be re-evaluated every frame.

>  I thought about using animate for delaying visual appearance, but there's not support for Step

Maybe you can use a transition, or an animation with a `delay`

**Reply by yanshay** [reply]
Thanks, I used Transitions together with delay to achieve what I needed.
